Enhabce oVirt provider create, view/edit credentials help text fields

sgratch opened the pull request:
Reference: https://github.com/kubev2v/forklift-console-plugin/issues/646
Reference: https://github.com/kubev2v/forklift-console-plugin/issues/541

Rephrase all oVirt fields help text messages and add hints for the following dialogs:
1. Create oVirt provider,
2. oVirt Credentials view/edit modes.

The main changes include:
1. Rephrase help text messages for `user`, `password`, `insecureSkipVerify`, `cacert` fields (`name` and `url` fields were already rephrased in separate PRs)
2. Support 'warning' validation state for the '`user`' field.
3. Add hint sections for `insecureSkipVerify`, `cacert` fields.
4. In Create/edit dialogs, hide the '`cacert`' field help text message and the hint section if `insecureSkipVerify`=true.

This fix assumes that leaving the '`cacert`' field empty, will trigger verifying the the system CA certificates.

### Screenshots

#### Create 
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/2fc6562a-1465-4640-a91f-6b4ee0a7fbd0)
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/d9495f69-684a-4375-9450-484766228185)

#### View Credentials
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/b5ac0456-5182-4e9a-b04f-18777f4e366c)

#### Edit Credentials
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/cc823fa1-68a9-43da-b305-4b9a4038df79)

yaacov commented:
can you add a little space above the blue hint box ?

![screenshot-user-images githubusercontent com-2023 11 21-18_03_09](https://github.com/kubev2v/forklift-console-plugin/assets/2181522/3f0fa44a-87c3-43c4-a18d-f6f04b1c4e9f)

sgratch commented:
> can you add a little space above the blue hint box ?
> 
@yaacov Done, please check the new screenshot added to the description comment

sgratch commented:
cc:// @RichardHoch @anarnold97 
Please review text changes

sgratch commented:
> This fix assumes that leaving the 'cacert' field empty, will trigger verifying the the system CA certificates.


@ahadas 
Do we still support the combination of unchecking the 'skip cert validation' and  leaving the oVirt provider's CA field empty for retrieving the system engine CA certificate by the webhook? 

Based on https://redhat-internal.slack.com/archives/C049WBVTRD2/p1694592350447549?thread_ts=1694517736.539569&cid=C049WBVTRD2, there was an issue with that. It's currently not working for me as well. What is the status of this?


@RichardHoch 
If above issue is still supported and should work then we need to document it as well since currently this option is not mentioned for the oVirt provider.

yaacov commented on packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/edit/OvirtCredentialsEdit.tsx:
```diff
@@ -165,6 +232,24 @@ export const OvirtCredentialsEdit: React.FC<EditComponentProps> = ({ secret, onC
+        <Hint>
+          <HintBody>
+            {cacertHelperTextMsgs.hint}
+            <Button
```
please move the button inside the hint msg

sgratch replied:
For supporting a href link, I can't add the link text into the text message, but rather just replace `Button` with `ExternalLink` and wrap both with the `Trans` component.
The text message split is still required, please see the fix. 
Any other idea?

yaacov commented on packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/edit/OvirtCredentialsEdit.tsx:
```diff
@@ -165,6 +232,24 @@ export const OvirtCredentialsEdit: React.FC<EditComponentProps> = ({ secret, onC
           isDisabled={insecureSkipVerify}
         />
       </FormGroup>
+      {insecureSkipVerify ? (
```
why do we need this check ?
do you want to hide the hint in some cases, what are this cases ? is there a reason to hide it in this cases ?

sgratch replied:
Yes, just to hide this hint in case of ignoring the ca - since there is no need for a hint of how to set this ca in that case. Make the dialog less loaded with information.

yaacov commented on packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/list/OvirtCredentialsList.tsx:
```diff
@@ -60,6 +79,13 @@ export const OvirtCredentialsList: React.FC<ListComponentProps> = ({ secret, rev
           )}
         </div>
+        <div className="forklift-page-secret-hint-div">
+          {key === 'insecureSkipVerify' ? (
```
why do you need this test ?
do you want to hide the hint ? why ? in what cases do you want to hide the hint ?

sgratch replied:
Yes, just to hide this hint in case of ignoring the ca - since there is no need for a hint of how to set this ca in that case. Make the dialog less loaded with information.

ahadas commented:
> > This fix assumes that leaving the 'cacert' field empty, will trigger verifying the the system CA certificates.
> 
> @ahadas Do we still support the combination of unchecking the 'skip cert validation' and leaving the oVirt provider's CA field empty for retrieving the system engine CA certificate by the webhook?
> 
> Based on https://redhat-internal.slack.com/archives/C049WBVTRD2/p1694592350447549?thread_ts=1694517736.539569&cid=C049WBVTRD2, there was an issue with that. It's currently not working for me as well. What is the status of this?

it is supposed to work once the certificate is added to the system certification pool but we didn't test it

yaacov commented on packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/edit/OvirtCredentialsEdit.tsx:
```diff
@@ -29,12 +33,58 @@ export const OvirtCredentialsEdit: React.FC<EditComponentProps> = ({ secret, onC
+  const insecureSkipVerifyHelperTextMsgs = {
+    error: t('Error: this field must be set to a boolean value.'),
+    success: t("If true (check box is checked), the provider's CA certificate won't be validated."),
+    hint: t(
```
the hint is not a validation help message, e.g. success, warninig, error, it's a hint that is always the same, regardless of the validation state.
please create a new const for it.

sgratch replied:
The const name `insecureSkipVerifyHelperTextMsgs` doesn't include the word 'verify' so I see it as a collection of text messages for this field and this specific hint is relevant for the `insecureSkipVerify` field only (explaining the enable/disable values).

But if you think that separating the help text based on component appearance is better then it's ok as well.

yaacov replied:
yes, the idea is that you can do `insecureSkipVerifyHelperTextMsgs[ callVerificationMethod(text) ]` in cases it makes sense, do the keys of the text array should be the same as the verification options.

yaacov commented on packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/edit/OvirtCredentialsEdit.tsx:
```diff
@@ -165,6 +236,21 @@ export const OvirtCredentialsEdit: React.FC<EditComponentProps> = ({ secret, onC
+      ) : (
+        <Hint>
+          <HintBody>
+            <Trans t={t} ns="plugin__forklift-console-plugin">
```
trans should include the text, this should look:
`<HintBody>{hintBody}</HintBody>`

yaacov commented on packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/list/OvirtCredentialsList.tsx:
```diff
@@ -13,19 +20,31 @@ export const OvirtCredentialsList: React.FC<ListComponentProps> = ({ secret, rev
+      description: t(
+        "If true (check box is checked), the provider's CA certificate won't be validated.",
+      ),
+      hint: t(
```
hint is not specific to specific a field, hint is a form thing, please create a new const for the hint text, or just use it inline

yaacov commented:
I see that mixing #646 for the help msgs and #541for the hints is not easy, lets separate the PR to one that do help msgs for the tasks in #646 and another for adding hints as required by #541, this way way we can focus better on each task

sgratch commented:
- After having an offline discussion, it was decided to convert both `hints` into  `tooltips` that will include relevant fields help text for now. 

  Reminder: It seems that that help text included relevant info for explaining how to use the fields, but since it's an extra info then tooltips seems preferable and less space expending. 
We can discuss using hints in the separate #541 issue. 

- This current fix addressed all comments mentioned above.

- I also updated the main comment and screenshots with the changes
